### PR TITLE
Using contact names in place of tx address when applicable

### LINKED ISSUE #7967: Address book names should be used in tx history
Currently if you've sent to an address that is in your address book and also has an ENS name, it defaults to using the ENS name.

We should probably prioritize displaying local pet-names for addresses first when available.

Screen Shot 2020-01-31 at 12 28 38 PM

### PR DESCRIPTION
**Pull request by ryanml**
Fixes: https://github.com/MetaMask/metamask-extension/issues/7967

I used another address of mine as a contact and `vitalik.eth` for the other

Screen Shot 2020-02-01 at 3 40 53 PM
Screen Shot 2020-02-01 at 3 41 51 PM
Screen Shot 2020-02-01 at 3 41 58 PM
Screen Shot 2020-02-01 at 3 42 35 PM

### CODE REVIEW

File: ui/app/components/app/transaction-list-item-details/transaction-list-item-details.component.js [hunk 1]
```diff
@@ -70,6 +71,14 @@ export default class TransactionListItemDetails extends PureComponent {
     onRetry(id)
   }
 
+  getNickName = address => {
```
Comment (whymarrh) [inline ui/app/components/app/transaction-list-item-details/transaction-list-item-details.component.js]: Can we move this logic into the container?

File: ui/app/components/app/transaction-list-item-details/transaction-list-item-details.component.js [hunk 2]
```diff
@@ -35,6 +35,7 @@ export default class TransactionListItemDetails extends PureComponent {
     rpcPrefs: PropTypes.object,
     senderAddress: PropTypes.string.isRequired,
     tryReverseResolveAddress: PropTypes.func.isRequired,
+    addressBook: PropTypes.array,
```
Comment (whymarrh) [inline ui/app/components/app/transaction-list-item-details/transaction-list-item-details.component.js]: As noted below, can we move the logic for receiving `senderNickname` into the container? Then we can change this to `senderNickname` and make it required.

Comment (whymarrh) [reply ui/app/components/app/transaction-list-item-details/transaction-list-item-details.component.js]: For the tests we can pass in whatever silly string

**Review by whymarrh — CHANGES_REQUESTED** [review]

File: ui/app/components/app/transaction-list-item-details/transaction-list-item-details.component.js (hunk 2 above)
Comment (ryanml) [reply ui/app/components/app/transaction-list-item-details/transaction-list-item-details.component.js]: @whymarrh nickname logic is now in the container, `recipientNickname` would also in this case need to be a required prop with this change. I think this is what you meant, let me know!

**Review by whymarrh — APPROVED** [review]
LGTM, thanks @ryanml!

**Pull request merged**